PhotoAlbumSample/app/src/androidTest/java/com/amazonaws/android/samples/photosharing/AppSyncMutationUITest.java

```diff
@@ -0,0 +1,264 @@
+//        UIActionsUtil.signOut();
+        final CountDownLatch latch = new CountDownLatch(1);
+        AWSMobileClient.getInstance().initialize(appContext, new Callback<UserStateDetails>() {
+            @Override
+            public void onResult(UserStateDetails result) {
+                latch.countDown();
+            }
+
+            @Override
+            public void onError(Exception e) {
+                latch.countDown();
+            }
+        });
+        latch.await();
+
+        final AWSConfiguration awsConfiguration = AWSMobileClient.getInstance().getConfiguration();
+
+        // AWS Configuration file should have AppSync configured.
+        // Should be generated by Amplify CLI.
+        assertNotNull(awsConfiguration.optJsonObject("AppSync"));
+        Log.e(TAG, "AppSync has been configured.");
+
+        // Check if goes back to LoginActivity
+        int timeOut = 0;
+        while (timeOut < MAX_TIME_OUT) {
+            try {
+                ViewInteraction button2 = onView(allOf(withId(R.id.user_pool_sign_in_view_id)));
+                button2.check(matches(isDisplayed()));
+                Log.e(TAG,"The view has gone back to LoginActivity.");
+                break;
+            } catch (NoMatchingViewException e) {
+
+            }
+
+            Thread.sleep(5000);
+            timeOut += 5000;
+        }
+
+        // Do sign in first
+        UIActionsUtil.signIn(UIActionsUtil.getUsername(), UIActionsUtil.getPassword());
+
+        // Check if user successfully signed in.
+        AWSMobileClient.getInstance().confirmSignIn("signInChallengeResponse", new Callback<SignInResult>() {
+            @Override
+            public void onResult(SignInResult result) {
+                Log.e(TAG, "SignInResult: " + result);
+                assertEquals(result.getSignInState().toString(), "DONE");
+            }
+
+            @Override
+            public void onError(Exception e) {
+                Log.e(TAG, "SignInResult Error: " + e);
+            }
+        });
+
+        // Check if it jumps into AlbumActivity
+        timeOut = 0;
+        while (timeOut < MAX_TIME_OUT) {
+            try {
+                onView(allOf(withText("AlbumActivity"), childAtPosition(allOf(withId(R.id.action_bar), childAtPosition(
+                        withId(R.id.action_bar_container),
+                        0)), 0), isDisplayed()));
+                break;
+                //view is displayed logic
+            } catch (NoMatchingViewException e) {
+                //view not displayed logic
+            } finally {
+                Thread.sleep(5000);
+                timeOut += 5000;
+            }
+        }
+
+        UIActionsUtil.addAlbum("TestAppSync");
+
+    }
+
+    @Test
+    public void createAndDeleteTest() throws InterruptedException {
+
+        UIActionsUtil.clickEdit();
+
+        // Check if the album is added successfully
+        onView(withText("TestAppSync")).check(matches(isDisplayed()));
```

Comment:
Is "TestAppSync" part of the resources strings such as `R.values.string1`? If so use them or declare a constant in this class

Reply:
Make a change to declare constants in this class.